controller/src/beerocks/master/db/db.cpp

```diff
@@ -2418,6 +2479,431 @@ const std::list<sChannelScanResults> &db::get_channel_scan_results(const sMacAdd
+        return false;
+    }
+
+    LOG(DEBUG) << __FUNCTION__ << ", selected_band = " << int(selected_bands);
+
+    auto timestamp = std::chrono::steady_clock::now();
+    if (save_to_persistent_db) {
+        LOG(DEBUG) << __FUNCTION__
+                   << ", configuring persistent-db, selected_bands = " << int(selected_bands);
+
+        std::unordered_map<std::string, std::string> values_map;
+        values_map.at(timestamp_str)      = timestamp_to_string_seconds(timestamp);
+        values_map.at(selected_bands_str) = std::to_string(selected_bands);
+
+        // update the persistent db
+        if (!update_client_entry_in_persistent_db(mac, values_map)) {
+            LOG(ERROR) << "failed to update client entry in persistent-db to for " << mac;
+            return false;
+        }
+    }
+
+    n->client_selected_bands       = selected_bands;
+    n->client_parameters_last_edit = timestamp;
+
+    return true;
+}
+
+beerocks::eFreqType db::get_client_selected_bands(const sMacAddr &mac)
+{
+    auto n = get_node(mac);
+    if (!n) {
+        LOG(ERROR) << "node not found for mac " << mac;
+        return beerocks::eFreqType::FREQ_UNKNOWN;
+    } else if (n->get_type() != beerocks::TYPE_CLIENT) {
+        LOG(ERROR) << __FUNCTION__ << "node " << mac << " is not a valid client!";
+        return beerocks::eFreqType::FREQ_UNKNOWN;
+    }
+
+    return n->client_selected_bands;
+}
+
+bool db::clear_client_persistent_db(const sMacAddr &mac)
+{
+    auto n = get_node(mac);
+    if (!n) {
+        LOG(ERROR) << "node not found for mac " << mac;
+        return false;
+    } else if (n->get_type() != beerocks::TYPE_CLIENT) {
+        LOG(ERROR) << __FUNCTION__ << "node " << mac << " is not a valid client!";
+        return false;
+    }
+
+    LOG(DEBUG) << __FUNCTION__ << ", setting client " << mac << " runtime info to default values";
+
+    n->client_parameters_last_edit  = std::chrono::steady_clock::time_point::min();
+    n->client_time_life_delay_sec   = std::chrono::seconds::zero();
+    n->client_stay_on_initial_radio = ePersistentParamBool::NOT_CONFIGURED;
+    n->client_initial_radio         = network_utils::ZERO_MAC;
+    n->client_stay_on_selected_band = ePersistentParamBool::NOT_CONFIGURED;
+    n->client_selected_bands        = beerocks::eFreqType::FREQ_UNKNOWN;
+
+    LOG(DEBUG) << __FUNCTION__ << ", removing client " << mac << " from persistent db";
+    auto db_entry        = client_db_entry_from_mac(mac);
+    auto type_client_str = type_to_string(beerocks::eType::TYPE_CLIENT);
+    if (!bpl::db_has_entry(type_client_str, db_entry)) {
+        LOG(DEBUG) << "client entry does not exist in persistent-db for " << db_entry;
+        return true;
+    }
+
+    if (!bpl::db_remove_entry(type_client_str, db_entry)) {
+        LOG(ERROR) << "failed to remove client entry " << db_entry;
+        return false;
+    }
+
+    return true;
+}
+
+bool db::update_client_persistent_db(const sMacAddr &mac)
+{
+    auto n = get_node(mac);
```

Comment:
```suggestion
    auto node = get_node(mac);
```

Reply:
replaced all new-code occurrences of `n` to `node`